src/schemathesis/specs/openapi/parameters.py

```diff
@@ -0,0 +1,348 @@
+
+    def transform_keywords(self, schema: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform Open API specific keywords into JSON Schema compatible form."""
+        definition = to_json_schema_recursive(schema, self.nullable_field)
+        # Headers are strings, but it is not always explicitly defined in the schema. By preparing them properly we
+        # can achieve significant performance improvements for such cases.
+        # For reference (my machine) - running a single test with 100 examples with the resulting strategy:
+        #   - without: 4.37 s
+        #   - with: 294 ms
+        #
+        # It also reduces the number of cases when the "filter_too_much" health check fails during testing.
+        if self.is_header:
+            definition.setdefault("type", "string")
+        return definition
+
+    def from_open_api_to_json_schema(self, open_api_schema: Dict[str, Any]) -> Dict[str, Any]:
+        """Convert Open API's `Schema` to JSON Schema."""
+        return {
+            key: value
+            for key, value in open_api_schema.items()
+            # Allow only supported keywords or vendor extensions
+            if key in self.supported_jsonschema_keywords or key.startswith("x-")
+        }
+
+
+class OpenAPI20Parameter(OpenAPIParameter):
+    """Open API 2.0 parameter.
+
+    https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameterObject
+    """
+
+    example_field = "x-example"
+    examples_field = "x-examples"
+    nullable_field = "x-nullable"
+    # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schemaObject
+    # Excluding informative keywords - `title`, `description`, `default`
+    # And `required`, because it has a different meaning here. It determines whether ot not this parameter is required
+    # or optional, which is not relevant because these parameters are later constructed into an "object" schema,
+    # and the value of this keyword is used there.
+    supported_jsonschema_keywords = (
+        "$ref",
+        "format",
+        "multipleOf",
+        "multipleOf",
+        "maximum",
+        "exclusiveMaximum",
+        "minimum",
+        "exclusiveMinimum",
+        "maxLength",
+        "minLength",
+        "pattern",
+        "maxItems",
+        "minItems",
+        "uniqueItems",
+        "maxProperties",
+        "minProperties",
+        "enum",
+        "type",
+        "items",
+        "allOf",
+        "properties",
+        "additionalProperties",
+    )
+
+    @property
+    def is_header(self) -> bool:
+        return self.location == "header"
+
+
+class OpenAPI30Parameter(OpenAPIParameter):
+    """Open API 3.0 parameter.
+
+    https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#parameter-object
+    """
+
+    example_field = "example"
+    examples_field = "examples"
+    nullable_field = "nullable"
+    # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#schema-object
```

Comment:
Needs clarification, that in contrast with Open API 2, the schema is separated here in the `schema` keyword